server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/period/Period.java

```diff
@@ -37,8 +37,11 @@
   private final long snapshotDate;
   private final String analysisUuid;
 
-  public Period(int index, String mode, @Nullable String modeParameter,
-    long snapshotDate, String analysisUuid) {
+  /**
+   * @deprecated replaced by {@link #Period(String, String, long, String)}
+   */
+  @Deprecated
+  public Period(int index, String mode, @Nullable String modeParameter, long snapshotDate, String analysisUuid) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)